rest/src/main/java/org/jboss/pnc/rest/notifications/model/NotificationFactory.java

```diff
@@ -0,0 +1,34 @@
+package org.jboss.pnc.rest.notifications.model;
+
+import org.jboss.pnc.spi.BuildStatus;
+import org.jboss.pnc.spi.events.BuildStatusChangedEvent;
+
+import javax.enterprise.context.ApplicationScoped;
+import java.util.HashMap;
+import java.util.Map;
+
+@ApplicationScoped
+public class NotificationFactory {
+
+    private Map<BuildStatus, NotificationEventType> externalEvents = new HashMap<>();
+
+    public NotificationFactory() {
+        externalEvents.put(BuildStatus.REPO_SETTING_UP, NotificationEventType.BUILD_STARTED);
```

Comment:
Ouch, in my opinion this BPM part should be refactored then :(

The same problem - when you just rename a constant in Backend the BPM 
will blow up. Moreover, since you are passing event name as a string in 
callback URL you won't get any error during compile time. It will blow 
up in runtime. I would like to avoid this kind of problems...



On 07/02/2015 10:52 AM, Matej Lazar wrote:
>
> In 
> rest/src/main/java/org/jboss/pnc/rest/notifications/model/NotificationFactory.java 
> <https://github.com/project-ncl/pnc/pull/425#discussion_r33758148>:
>
> > +package org.jboss.pnc.rest.notifications.model;
> > +
> > +import org.jboss.pnc.spi.BuildStatus;
> > +import org.jboss.pnc.spi.events.BuildStatusChangedEvent;
> > +
> > +import javax.enterprise.context.ApplicationScoped;
> > +import java.util.HashMap;
> > +import java.util.Map;
> > +
> > +@ApplicationScoped
> > +public class NotificationFactory {
> > +
> > +    private Map<BuildStatus, NotificationEventType> externalEvents = new HashMap<>();
> > +
> > +    public NotificationFactory() {
> > +        externalEvents.put(BuildStatus.REPO_SETTING_UP, NotificationEventType.BUILD_STARTED);
>
> I find it more confusing having another status for the same thing. We 
> already have BuildStatus, BuildSetStatus and BuildStatusChangedEvent, 
> BuildSetStatusChangedEvent as part of SPI and already used in the REST.
> https://github.com/project-ncl/pnc/blob/33ebb8c8d7dd2594f878ac1f5d1ed0cdb3c73009/rest/src/main/java/org/jboss/pnc/rest/trigger/BuildTriggerer.java#L102
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/project-ncl/pnc/pull/425/files#r33758148>.
>